### PR TITLE
fix: Cannot specify more than one extra_container_definitions

### PR DESCRIPTION
**Pull request by dynamike**
## Description
Resolves issues defined in https://github.com/terraform-aws-modules/terraform-aws-atlantis/issues/269 when trying to pass in multiple container definitions into `extra_container_definitions`. The main issue is Terraform's requirement for exact matching objects when untyping a variable as `any` More detail is defined in https://github.com/hashicorp/terraform/issues/26265. For the time being, this PR untypes the variable to get around the error. A more robust solution would leverage required and optional attributes to define a container definition. 

## Motivation and Context
Resolves https://github.com/terraform-aws-modules/terraform-aws-atlantis/issues/269, which seems like an increasingly common issue with more and more sidecar containers used for things like log routing and observability.

## Breaking Changes
There shouldn't be any breaking changes. 

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 0.13.1"
+  required_version = ">= 1.3.0"
```
Comment (bryantbiggs) [inline versions.tf:2]: This is a breaking change and we do not want to push users to the bleeding edge as it will reduce adoption

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
I think we can provide a solution without the need for a breaking change and pushing users to use the latest version of Terraform

**Review by bryantbiggs — APPROVED** [review]

**Comment by bryantbiggs** [comment]
Thank you @dynamike

**Comment by antonbabenko** [comment]
This PR is included in version 3.22.1 :tada:

**Pull request merged**